### ISSUE TITLE
com.hazelcast.internal.util.ModularJavaUtilsTest.testNoMissingPackageAccessInTestsuite [HZ-668]

### ISSUE THREAD

**Issue by slavagithub** [hazelcast/hazelcast#19874]
Master (rev. e817a03)

Failed on Zulu, JDK17: http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-ZuluJDK17/5/testReport/

<details><summary>Stacktrace:</summary>

```
org.mockito.exceptions.verification.NeverWantedButInvoked: 

iLogger.warning(<any string>);
Never wanted here:
-> at com.hazelcast.internal.util.ModularJavaUtilsTest.testNoMissingPackageAccessInTestsuite(ModularJavaUtilsTest.java:58)
But invoked here:
-> at com.hazelcast.internal.util.ModularJavaUtils.checkPackageRequirements(ModularJavaUtils.java:93)

	at com.hazelcast.internal.util.ModularJavaUtilsTest.testNoMissingPackageAccessInTestsuite(ModularJavaUtilsTest.java:58)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
```

</details>

<details><summary>Standard output:</summary>

```
Started Running Test: testWriteRead
Started Running Test: testBefore
```

</details>

**Comment by ldziedziul** [comment]
Fixed with https://github.com/hazelcast/hazelcast/pull/20295

**Comment by slavagithub** [comment]
It came back in Openj9-16 - http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Openj9-16/45/testReport/com.hazelcast.internal.util/ModularJavaUtilsTest/testNoMissingPackageAccessInTestsuite/, reopening

**Comment by olukas** [comment]
Failed also on Openj9-11: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Openj9-11/64/testReport/com.hazelcast.internal.util/ModularJavaUtilsTest/testNoMissingPackageAccessInTestsuite/

**Comment by olukas** [comment]
Failed on OpenJ9-17: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Openj9-17/1/testReport/com.hazelcast.internal.util/ModularJavaUtilsTest/testNoMissingPackageAccessInTestsuite/

**Comment by slavagithub** [comment]
failed on Openj9-16 - http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Openj9-16/48/testReport/com.hazelcast.internal.util/ModularJavaUtilsTest/testNoMissingPackageAccessInTestsuite/

**Comment by olukas** [comment]
It still fails on OpenJ9-16: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Openj9-16/49/testReport/com.hazelcast.internal.util/ModularJavaUtilsTest/testNoMissingPackageAccessInTestsuite/

**Comment by olukas** [comment]
Reopening since it happened on `5.1.z` branch on Openj9-11: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-5.maintenance-Openj9-11/25/testReport/com.hazelcast.internal.util/ModularJavaUtilsTest/testNoMissingPackageAccessInTestsuite/

**Comment by vbekiaris** [comment]
fixed on 5.1.z branch by #20959
